src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -667,7 +663,7 @@ private void checkBaseTypes(DetailAST clause) {
             if (child.getType() == TokenTypes.IDENT) {
                 checkIdent(child);
             }
-            else if (child.getType() == TokenTypes.TYPE_ARGUMENTS) {
+            else {
```

Comment:
The code inside the if statement is checking the children of the child token. When this token is not type arguments, it is specifically comma tokens which have no children. The code in the if is examining the children of the child token, which makes it not matter what is given to it since they have no children. If something ever does have children, we are looking for a very specific child which again contributes to that it will never been found.